queenbee/schema/workflow.py

```diff
@@ -77,6 +101,30 @@ def nodes_links(self):
         return {'nodes': nodes, 'links': links}
 
 
+def list_artifacts(obj):
+    artifacts = []
+    
+    if 'flow' in obj:
+        for dag in obj['flow']:
+            artifacts += list_artifacts(dag)
+    
+    if 'templates' in obj:
+        for template in obj['templates']:
+            artifacts += list_artifacts(template)
+    
+    if isinstance(obj, DAG):
+        for task in obj.tasks:
+            if 'arguments' in task and 'artifacts' in task.arguments:
+                artifacts = artifacts + task.arguments.artifacts  
+    
+    if isinstance(obj, Function):
+        if obj.inputs != None and obj.inputs.artifacts != None:
+            artifacts = artifacts + obj.inputs.artifacts
+        if obj.outputs != None and obj.outputs.artifacts != None:
+            artifacts = artifacts + obj.outputs.artifacts
```

Comment:
This will overwrite the input artifacts if any. Is this expected?

Reply:
Yep, artifacts is just an empty list created at the start of this function. It shouldn't change anything in the schema (unless I'm misreading my own code here).

Reply:
but if inputs and outputs both have artifacts don't you want to collect them all?

```python
        artifacts = []
        if obj.inputs != None and obj.inputs.artifacts != None:
            artifacts.extend(artifacts + obj.inputs.artifacts)
        if obj.outputs != None and obj.outputs.artifacts != None:
            artifacts.extend(artifacts + obj.outputs.artifacts)
```